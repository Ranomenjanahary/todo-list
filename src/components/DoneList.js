
export const DoneList = ({ done }) => (
    <div className="done">
      <h3>DONE</h3>
      <ul>
        {done.map((todo, index) => (
          <li key={index}>{todo}</li>
        ))}
      </ul>
    </div>
  );