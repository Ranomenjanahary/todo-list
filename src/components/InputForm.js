import React from 'react';

export const InputForm = ({ handleSubmit, todo, setTodo }) => (
  <form onSubmit={handleSubmit}>
    <input
      type="text"
      value={todo}
      onChange={(e) => setTodo(e.target.value)}
      onKeyDown={(e) => {
        e.key === 'Enter' && handleSubmit();
      }}
      placeholder="Add task"
    />
  </form>
);