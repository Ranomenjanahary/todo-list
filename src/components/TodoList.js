
export const TodoList = ({ doing, handleCheck }) => (
    <div className="todo">
      <h3>TODO</h3>
      <ul>
        {doing.map((todo, index) => (
          <li key={index}>
            {todo}
            <input type="checkbox" onClick={() => handleCheck(index)} />
          </li>
        ))}
      </ul>
    </div>
  );