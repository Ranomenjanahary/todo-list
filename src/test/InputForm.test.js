import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { InputForm } from '../components/InputForm';


describe('InputForm component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<InputForm handleSubmit={() => {}} todo="" setTodo={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls handleSubmit on form submit', () => {
    const handleSubmit = jest.fn();
    const { getByPlaceholderText } = render(
      <InputForm handleSubmit={handleSubmit} todo="" setTodo={() => {}} />
    );
    const input = getByPlaceholderText('Add task');
    fireEvent.submit(input.closest('form'));
    expect(handleSubmit).toHaveBeenCalled();
  });

  it('updates todo state on input change', () => {
    const setTodo = jest.fn();
    const { getByPlaceholderText } = render(
      <InputForm handleSubmit={() => {}} todo="" setTodo={setTodo} />
    );
    const input = getByPlaceholderText('Add task');
    fireEvent.change(input, { target: { value: 'new todo' } });
    expect(setTodo).toHaveBeenCalledWith('new todo');
  });

  it('calls handleSubmit on Enter key down', () => {
    const handleSubmit = jest.fn();
    const { getByPlaceholderText } = render(
      <InputForm handleSubmit={handleSubmit} todo="" setTodo={() => {}} />
    );
    const input = getByPlaceholderText('Add task');
    fireEvent.keyDown(input, { key: 'Enter', code: 13 });
    expect(handleSubmit).toHaveBeenCalled();
  });
});
