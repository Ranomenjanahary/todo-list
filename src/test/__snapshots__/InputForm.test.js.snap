// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm component matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="Add task"
      type="text"
      value=""
    />
  </form>
</DocumentFragment>
`;
